src/prefect/tasks/secrets/vault_secret.py

```diff
@@ -0,0 +1,161 @@
+import re
+import json
+import os
+
+try:
+    import hvac
+except ImportError as err:
+    raise ImportError(
+        "Using `prefect.tasks.secrets.vault_secret` requires hvac to be installed."
+    ) from err
+
+import prefect
+from prefect.tasks.secrets import SecretBase, PrefectSecret
+from prefect.utilities.tasks import defaults_from_attrs
+from prefect.exceptions import ClientError
+
+
+class VaultSecret(SecretBase):
+    """
+    Vault Secrets Task.  This task retrieves a secret using the Hashicorp Vault service.
+    Local secrets are checked before querying the Vault service.
+
+    Client configuration:
+        - vault_addr: set via env var.  Both VAULT_ADDR and vault_addr supported
+
+    Args:
+        - name (str): The secret name defined by the Vault secret path "<mount-point>/<path>"
+        - vault_credentials_secret (str, optional): name of the PrefectSecret containing Vault
+            credentials.
+                Defaults to a PrefectSecret named `VAULT_CREDENTIALS`.
+                Supported vault client authentication methods:
+                * token { 'VAULT_TOKEN: '<token>' }
+                * appRole: { 'VAULT_ROLE_ID': '<role-id>',
+                             'VAULT_SECRET_ID': '<secret-id>' }
+        - **kwargs (Any, optional): additional keyword args passed to the Task constructor
+
+    Raises:
+        - ValueError: if a `result` keyword is passed
+    """
+
+    def __init__(
+        self, name: str, vault_credentials_secret: str = "VAULT_CREDENTIALS", **kwargs
+    ):
+        self.vault_credentials_secret = vault_credentials_secret
+        kwargs["name"] = name
+        super().__init__(**kwargs)
+
+    def _get_vault_secret(self, name: str):
+        """
+        Connect to vault service and retrieve a secret value
+        Arguments:
+            - name (str): name of the secret to be retrieved using "<mount_point>/<path>"
+
+        Returns:
+            - secret (dict): a dict of the secret key/value items retrieved
+
+        Raises:
+            - ClientError: unable to configure the vault client
+            - PermissionError: unable to authenticate or attempting an unsupported auth method
+            - KeyError: unable to lookup secret in vault using the secret path
+            - PermissionError: provided token/role not authorised to access the secret
+        """
+        self.logger.debug(f"looking up vault path: {name}")
+        client = hvac.Client()
+        # get vault address url
+        vault_url = os.getenv("VAULT_ADDR", default=None)
+        if vault_url is None:
+            vault_url = os.getenv("vault_addr", default=None)
+            if vault_url is None:
+                raise ClientError(
+                    "VAULT_ADDR url var not found. "
+                    'Either "VAULT_ADDR" or "vault_addr" env var required.'
+                )
+        client.url = vault_url
+        self.logger.debug(f"vault addr set to: {client.url}")
```

Comment:
```suggestion
        self.logger.debug(f"Vault addr set to: {client.url}")
```
Generally, I'd capitalize the first word of each of these debug logs to match the rest of the logs from Prefect

Reply:
debug messages updated to use consistent style.